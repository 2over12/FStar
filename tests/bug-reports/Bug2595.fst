module Bug2595

noeq
type sum_type  =
  | SumType1: string -> sum_type
  | SumType2: nat -> sum_type

let test_buggy (x:(b:bool & (if b then nat else string))): sum_type =
  match x with
  | (|false, x2|) -> SumType1 x2
  | (|true, x4|) -> SumType2 x4
