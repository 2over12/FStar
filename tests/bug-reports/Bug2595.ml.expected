open Prims
type sum_type =
  | SumType1 of Prims.string 
  | SumType2 of Prims.nat 
let (uu___is_SumType1 : sum_type -> Prims.bool) =
  fun projectee ->
    match projectee with | SumType1 _0 -> true | uu___ -> false
let (__proj__SumType1__item___0 : sum_type -> Prims.string) =
  fun projectee -> match projectee with | SumType1 _0 -> _0
let (uu___is_SumType2 : sum_type -> Prims.bool) =
  fun projectee ->
    match projectee with | SumType2 _0 -> true | uu___ -> false
let (__proj__SumType2__item___0 : sum_type -> Prims.nat) =
  fun projectee -> match projectee with | SumType2 _0 -> _0
let (test_buggy : (Prims.bool, Obj.t) Prims.dtuple2 -> sum_type) =
  fun x ->
    match x with
    | Prims.Mkdtuple2 (false, x2) -> SumType1 (Obj.magic x2)
    | Prims.Mkdtuple2 (true, x4) -> SumType2 (Obj.magic x4)